tonic-build/src/client.rs

```diff
@@ -115,12 +117,14 @@ fn generate_server_streaming(method: &Method, proto: &str, path: String) -> Toke
     let (request, response) = crate::replace_wellknown(proto, &method);
 
     quote! {
-        pub async fn #ident(&mut self, request: tonic::Request<#request>)
-            -> Result<tonic::Response<tonic::codec::Streaming<#response>>, tonic::Status> {
+        pub async fn #ident(
+            &mut self,
+            request: impl Into<tonic::Request<#request>>,
```

Comment:
I'm on the fence as to whether these should use `Into` or if we should still add an `IntoRequest` trait with blanket impls. I _think_ I prefer just using `Into`, but having an `IntoRequest` trait might be more consistent with `IntoStreamingRequest`...

Reply:
I was having trouble with overlapping trait implementations for an unbounded T and Request<T> but I _think_ this works as a replacement for From/Into, without generating any code:

```rust
pub trait IntoRequest<T> {
    fn into_request(self) -> Request<T>;
}

impl<T> IntoRequest<T> for T {
    fn into_request(self) ->  Request<Self> {
        Request::new(self)
    }
}

impl<T> IntoRequest<T> for Request<T> {
    fn into_request(self) -> Request<T> {
        self
    }
}
```
We could do that too, unless @LucioFranco is not happy?

Reply:
I could go either way on this, honestly.

Reply:
I'd prefer to have both `IntoRequest` and `IntoStreamingRequest` personally.

Reply:
Done